### ISSUE TITLE
Can't add nodes in the editor to create a new scene (also can't add nodes to an existing scene)

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#55015]
*Related to https://github.com/godotengine/godot/issues/55014.*

### Godot version

3.5.beta.custom_build (458f5b42a63ca0f9b2199e21165354a1e9938e37)

### System information

Fedora 34, GeForce GTX 1080 (NVIDIA 470.74)

### Issue description

Trying to add a node to a blank or existing scene results in the following error being printed, and no node being added:

```
ERROR: Condition "!p_node->is_inside_tree()" is true.
   at: add_node (editor/editor_data.cpp:1028)
```

https://user-images.githubusercontent.com/180032/141872165-c37dd770-70dc-4883-94cb-fa2399ef1685.mp4

### Steps to reproduce

- Create a new project.
- Try to add a node using the Scene tree dock (either using a preset or the Custom node dialog).
- Nothing appears, but you can see an error message:

### Minimal reproduction project

N/A

**Comment by akien-mga** [comment]
Can't reproduce in 3ac2999f2, so this would be a very recent regression (from after my last batch of cherry-picks).

**Comment by akien-mga** [comment]
This is likely caused by #54433, CC @rxlecky @RandomShaper.

**Comment by akien-mga** [comment]
Fixed by #55022.
